torch/_inductor/constant_folding.py

```diff
@@ -195,3 +200,60 @@ def constant_fold(gm, constraint_fn: Optional[Callable[[torch.fx.Node], bool]] =
     gm.graph.eliminate_dead_code()
     gm.graph.lint()
     gm.recompile()
+
+
+@torch.utils._python_dispatch._disable_current_modes()
+def constant_graph_tag(gm: torch.fx.GraphModule):
+    cf = ConstantFolder(gm, skip_constructors=True)
+    cf.run()
+
+    for node in gm.graph.nodes:
+        if (
+            node.op == "get_attr"
+            or node in cf.node_replacements
+            or node in cf.replaced_uses
+        ):
+            node.tag = CONST_MODULE_TAG
+        else:
+            node.tag = MODULE_TAG
+
+
+def get_constant_graph(gm: torch.fx.GraphModule) -> torch.fx.GraphModule:
+    """
+    Construct a GraphModule which corresponds to the part which could be
+    constant folded in provided gm.
+    """
+
+    constant_graph_tag(gm)
+    # We rewrite the tags, if it's a constant being directly consumed, without
+    # any folding opportunity, we keep it in main gm.
+    for node in gm.graph.nodes:
+        if node.op == "get_attr":
+            used_to_fold = False
+            for u in node.users:
+                if u.tag == CONST_MODULE_TAG:
```

Comment:
I think this is correct because `CONST_MODULE_TAG` has been properly populated by `ConstantFolder`. Is there a good to test it? Maybe adding some logging and verify when testing?